sre-recipes/recipes/recipe_1/recipe_1.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# -*- coding: utf-8 -*-
+"""
+This module contains the implementation of recipe 1
+"""
+
+import os
+import subprocess
+from recipe import Recipe
+
+
+class Recipe1(Recipe):
+    """
+    This class implements recipe 1, which purposefully
+    introduces latency into the frontend service.
+    """
+
+    def __init__(self):
+        self.__service = 'frontend'
+
+    @staticmethod
+    def get_relative_path(relative_path, filename, extension):
+        """
+        Returns the absolute path to the filename given the relative
+        path to the file, filename, and file extension
+        """
+        dir_abs_path = os.path.dirname(os.path.abspath(__file__))
+        return dir_abs_path + relative_path + filename + extension
+
+    @staticmethod
+    def get_working_path(service):
```

Comment:
If you want a method to be "private" you can start it with an underscore, like _get_working_path

Reply:
Cool, good to know for future reference! As mentioned in previous comments, I don't think I'll need the pathfinding methods anymore.